pkg/validationfile/loader.go

```diff
@@ -52,124 +51,96 @@ func PopulateFromFiles(ds datastore.Datastore, filePaths []string) (*PopulatedVa
 // PopulateFromFilesContents populates the given datastore with the namespaces and tuples found in
 // the validation file(s) contents specified.
 func PopulateFromFilesContents(ds datastore.Datastore, filesContents map[string][]byte) (*PopulatedValidationFile, datastore.Revision, error) {
-	var revision datastore.Revision
-	var nsDefs []*core.NamespaceDefinition
+	var schema string
+	var objectDefs []*core.NamespaceDefinition
 	var caveatDefs []*core.CaveatDefinition
-	schema := ""
 	var tuples []*core.RelationTuple
+	var updates []*core.RelationTupleUpdate
+
+	var revision datastore.Revision
+
 	files := make([]ValidationFile, 0, len(filesContents))
 
+	// Parse each file into definitions and relationship updates.
 	for filePath, fileContents := range filesContents {
+		// Decode the validation file.
 		parsed, err := DecodeValidationFile(fileContents)
 		if err != nil {
 			return nil, datastore.NoRevision, fmt.Errorf("error when parsing config file %s: %w", filePath, err)
 		}
 
 		files = append(files, *parsed)
 
-		// Add schema-based namespace definitions.
+		// Disallow legacy sections.
+		if len(parsed.NamespaceConfigs) > 0 {
+			return nil, revision, fmt.Errorf("definitions must be specified in `schema`")
+		}
```

Comment:
Do you think folks out there could have validation files using legacy representation, and this will make it stop parsing? Do we have any recommendations for those folks to migrate to the new representation? (maybe a note in the release?

Reply:
I plan to add a note in the release and yes, that's why I expressly turned it into an error, although I'd be quite surprised if anyone is actually using the old format

Reply:
👍🏻